src/command.ts

```diff
@@ -308,3 +310,28 @@ export class Command {
     };
   }
 }
+
+// Project IDs must follow a certain format, as documented at:
+// https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects#resource:-project
+// However, the regex below, matching internal ones, is more permissive so that
+// some legacy projects with irregular project IDs still works.
+const PROJECT_ID_REGEX = /^(?:[^:]+:)?[a-z0-9-]+$/;
+
+/**
+ * Validate the project id and throw on invalid format.
+ * @param project the project id to validate
+ * @throws {FirebaseError} if project id has invalid format.
+ */
+export function validateProjectId(project: string): void {
+  if (PROJECT_ID_REGEX.test(project)) {
+    return;
+  }
+  track("error", "invalid-project-id");
+  const invalidMessage = "Invalid project id: " + clc.bold(project) + ".";
+  if (project.toLowerCase() !== project) {
```

Comment:
Is there a point in erroring out instead of just normalizing to lowercase on their behalf?

Reply:
My assumption is that if this is triggered, then mostly likely the developer has entered the project display name ("MyProject") rather than the project id, since GCP documentation is quite confusing here. We can definitely add a feature to convert it lowercase as a non-breaking change later if we see feature requests like that.